docs(root): add further installation instructions (#3892)
gchq/Bailo: README.md

#### Requirements

Follow the instructions below to get a local development environment running. You will need:

- Node v26 - [https://nodejs.org/en/download](https://nodejs.org/en/download)

> - Docker / Docker Compose - [https://www.docker.com/get-started/](https://www.docker.com/get-started/)

Reviewer: Prefer anchor text link to bare URL
```suggestion
- [Node v26](https://nodejs.org/en/download/current)
- [Docker / Docker Compose](https://www.docker.com/get-started/)
```

#### Development Build

To run in development mode (modified files on your host machine will be reloaded into the running application):

> [!NOTE]

> > You will need to install and start docker desktop before running these commands.

Reviewer 2: As this is specific to installing Docker on Windows and Mac I would suggest this should be an update to the [existing requirements section for local Docker development](https://github.com/gchq/Bailo/pull/3892/changes#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5L44), making clear the difference in installing Docker for Lunix vs Mac and Windows.

> > You will need to install and start docker desktop before running these commands.

Reviewer: This assumes Docker Desktop is the only supported Docker runtime, but Linux users may be running Docker Engine directly without Docker Desktop.
```suggestion
> Ensure Docker and Docker Compose are installed before running these commands.
```
